WordPress/Sniffs/VIP/ValidatedSanitizedInputSniff.php

```diff
@@ -122,8 +125,7 @@ public function process_token( $stackPtr ) {
 
 		// Check for validation first.
 		if ( ! $this->is_validated( $stackPtr, $array_key, $this->check_validation_in_scope_only ) ) {
-			$this->phpcsFile->addError( 'Detected usage of a non-validated input variable: %s', $stackPtr, 'InputNotValidated', $error_data );
-			// return; // Should we just return and not look for sanitizing functions ?
+			$notValidated = true;
 		}
 
 		if ( $this->has_whitelist_comment( 'sanitization', $stackPtr ) ) {
```

Comment:
Because of this if condition and the one below it, if sanitization isn't needed the validation error would no longer be given either. This logic needs to be updated to fix that. (And also the unit tests, so that they'd catch this.)